### DISCUSSION TITLE
Add support for semantic highlighting

### DISCUSSION THREAD

**Discussion by yunti** [giraud/reasonml-idea-plugin#123] [Feature request]
It's surpising how useful this is now I'm used to it.
https://twitter.com/intellijidea/status/844901295823585280

eg simple example,  reason
<img width="488" alt="image" src="https://user-images.githubusercontent.com/1826271/49764299-f4778d80-fcc6-11e8-9e49-8421df695453.png">

vs js
<img width="563" alt="image" src="https://user-images.githubusercontent.com/1826271/49764327-03f6d680-fcc7-11e8-8a26-faa05c6c033b.png">

**Comment by giraud** [comment]
interesting, but I don't know which version of Idea supports that feature.

**Comment by yunti** [comment]
The js screenshot is from intellij ultimate but this one is from pycharm, so it's probably supported throughout the platform. 

<img width="913" alt="image" src="https://user-images.githubusercontent.com/1826271/49791612-c034b900-fd28-11e8-96ce-1c91db762c81.png">

**Comment by giraud** [comment]
it's a christmas tree... how can spot anything ? :)

**Comment by yunti** [comment]
Yup not the best example - but just wanted to show something quickly.  In an actual editor it becomes must clearer very quickly and it's way faster to scan and pick out things.
And well, it's nearly christmas 🎄
